### PR TITLE
doc: fix listnodes schema, doc.

### PR DESCRIPTION
**Pull request by rustyrussell**
Shows the dangers of "additionalProperties": true.  We didn't have an else
clause, so our incorrect (and, with DF, incomplete!) schema was accepted.

I despair of getting anyone else to write a decent schema with these
semantics :(

Changelog-Fixed: doc: listnodes fields now correctly documented.
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>

### CODE REVIEW

**Comment by cdecker** [comment]
Seems to be missing a `required` key somewhere:

```
fromschema doc/lightning-listnodes.7.md
Traceback (most recent call last):
  File "tools/fromschema.py", line 276, in <module>
    main(parsed_args.schemafile, parsed_args.markdownfile)
  File "tools/fromschema.py", line 266, in main
    generate_from_schema(schema)
  File "tools/fromschema.py", line 234, in generate_from_schema
    output_members(sub)
  File "tools/fromschema.py", line 193, in output_members
    output_members(ifclause['then'], indent + '  ')
  File "tools/fromschema.py", line 140, in output_members
    if p in sub['required']:
KeyError: 'required'
make: *** [doc/Makefile:102: doc/lightning-listnodes.7.md] Error 1
```

File: doc/lightning-listnodes.7.md [hunk 1]
```diff
@@ -29,16 +29,24 @@ RETURN VALUE
...
     - **address** (string): address in expected format for *type*
     - **port** (u16): port number
+
+If **option_will_fund** is present:
+  - **lease_fee_base_msat** (msat): the fixed fee for a lease (whole number of satoshis)
+  - **lease_fee_basis** (u32): the proportional fee in basis points (parts per 10,000) for a lease
+  - **funding_weight** (u32): the onchain weight you'll have to pay for a lease
+  - **channel_fee_max_base_msat** (msat): the maximum base routing fee this node will charge during the lease
+  - **channel_fee_max_proportional_thousandths** (msat): the maximum proportional routing fee this node will charge during the lease (in thousandths, not millionths like channel_update)
+  - **compact_least** (hex, optional): the lease as represented in the node_announcement
```
Comment (niftynei) [inline doc/lightning-listnodes.7.md:49]:
```suggestion
  - **compact_lease** (hex, optional): the lease as represented in the node_announcement
```

File: doc/schemas/listnodes.schema.json [hunk 2]
```diff
@@ -70,6 +72,58 @@
...
+					},
+					"channel_fee_max_base_msat": {
+					    "type": "msat",
+					    "description": "the maximum base routing fee this node will charge during the lease"
+					},
+					"channel_fee_max_proportional_thousandths": {
+					    "type": "msat",
+					    "description": "the maximum proportional routing fee this node will charge during the lease (in thousandths, not millionths like channel_update)"
+					},
+					"compact_least": {
```
Comment (niftynei) [inline doc/schemas/listnodes.schema.json:120]:
```suggestion
					"compact_lease": {
```

**Review by niftynei — APPROVED** [review]

File: doc/schemas/listnodes.schema.json (hunk 2 above)
Comment (rustyrussell) [reply doc/schemas/listnodes.schema.json:120]: Gah, this was clearly untested... :(

**Comment by cdecker** [comment]
Regenerated the docs since it was failing with a different source hash.

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/ElementsProject/lightning/pull/4750/commits/a68cf25288000275420b20fa2ade9b132162a31a

**Pull request merged**